main.go

```diff
@@ -148,6 +153,14 @@ func run(ctx context.Context, ms *xmain.State) (err error) {
 	}
 	ms.Log.Debug.Printf("using theme %s (ID: %d)", match.Name, *themeFlag)
 
+	if *darkThemeFlag != math.MaxInt64 {
+		match = d2themescatalog.Find(*darkThemeFlag)
+		if match == (d2themes.Theme{}) {
+			return xmain.UsageErrorf("--dark_theme could not be found. The available options are:\n%s\nYou provided: %d", d2themescatalog.CLIString(), *darkThemeFlag)
```

Comment:
we should have a separate catalog of dark themes right? could be misleading to suggest a list with light themes for value of `dark_theme `

Reply:
nah, the two catalogs are merged, anyone can set for example the theme with id `0` a a primary theme and the theme with id `4` as a dark preference. I think this is a more canonical way as this will make it easier to load themes from files at runtime.

Reply:
is that alright?

Reply:
> anyone can set for example the theme with id 0 a a primary theme and the theme with id 4 as a dark preference.

why would anyone do that other than on accident though?

> I think this is a more canonical way as this will make it easier to load themes from files at runtime.

not sure what you mean here. how would it be easier?

Reply:
Well, because every theme should be categorized as a dark or a light one. Actually it affects the precompiled themes too.

Imagine what would happen when the user wanted to render only a dark theme SVG.

This way they would simply set the primary theme to a dark one (`--theme 200`).

Reply:
https://github.com/terrastruct/d2/issues/623#issuecomment-1374788565

Reply:
i don't follow what that has to do with separate catalogs though. i just mean 2 separate lists. using a dark theme for `theme` is fine, maybe they just like the dark mode for everyone. but using a light theme for `dark-theme` would always be a mistake.

> Imagine what would happen when the user wanted to render only a dark theme SVG.

to be clear i'm in agreement that users should be able to specify just a dark theme as their primary. i just want to give a better experience to users who want to know, "oh sweet D2 has dark themes! let's list them". it'd be nice if we had a list we could just print out, instead of giving them the full list and making them infer based on names

Reply:
so you want to still keep the `id`s? like that `2**` for dark themes etc.?

Reply:
yeah that's perfectly fine